pkg/scalers/splunk_observability_scaler_test.go

```diff
@@ -0,0 +1,89 @@
+package scalers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+)
+
+type parseSplunkObservabilityMetadataTestData struct {
+	metadata   map[string]string
+	authParams map[string]string
+	isError    bool
+}
+
+type SplunkObservabilityMetricIdentifier struct {
+	metadataTestData *parseSplunkObservabilityMetadataTestData
+	triggerIndex     int
+	metricName       string
+}
+
+var validSplunkObservabilityAuthParams = map[string]string{
+	"accessToken": "my-suyper-secret-access-token",
+	"realm":       "my-realm",
+}
+
+var validSplunkObservabilityMetadata = map[string]string{
+	"query":                 "data('demo.trans.latency').max().publish()",
+	"duration":              "10",
+	"targetValue":           "200.0",
+	"queryAggregator":       "avg",
+	"ActivationTargetValue": "1.1",
+}
+
+var testSplunkObservabilityMetadata = []parseSplunkObservabilityMetadataTestData{
+	// Valid metadata and valid auth params, pass.
+	{validSplunkObservabilityMetadata, validSplunkObservabilityAuthParams, false},
+	// no params at all, fail
+	{map[string]string{}, map[string]string{}, true},
+	// No meta dada but valid auth, fail.
+	{map[string]string{}, validSplunkObservabilityAuthParams, true},
+	// Valid meta dada but no auth params, fail.
+	{validSplunkObservabilityMetadata, map[string]string{}, true},
+	// Missing 'query' field, fail
+	{map[string]string{"duration": "10", "targetValue": "200.0", "queryAggregator": "avg", "activationTargetValue": "1.1"}, validSplunkObservabilityAuthParams, true},
+	// Missing 'duration' field, fail
+	{map[string]string{"query": "data('demo.trans.latency').max().publish()", "targetValue": "200.0", "queryAggregator": "avg", "activationTargetValue": "1.1"}, validSplunkObservabilityAuthParams, true},
+	// Missing 'targetValue' field, fail
+	{map[string]string{"query": "data('demo.trans.latency').max().publish()", "duration": "10", "queryAggregator": "avg", "activationTargetValue": "1.1"}, validSplunkObservabilityAuthParams, true},
+	// Missing 'queryAggregator' field, fail
+	{map[string]string{"query": "data('demo.trans.latency').max().publish()", "duration": "10", "targetValue": "200.0", "activationTargetValue": "1.1"}, validSplunkObservabilityAuthParams, true},
+	// Missing 'activationTargetValue' field, fail
+	{map[string]string{"query": "data('demo.trans.latency').max().publish()", "duration": "10", "targetValue": "200.0", "queryAggregator": "avg"}, validSplunkObservabilityAuthParams, true},
+}
+
+var SplunkObservabilityMetricIdentifiers = []SplunkObservabilityMetricIdentifier{
+	{&testSplunkObservabilityMetadata[0], 0, "demo-trans-latency"},
+	{&testSplunkObservabilityMetadata[0], 1, "demo-trans-latency"},
```

Comment:
I think that this test is failing because the metric name must contain the index (it's done by `GenerateMetricNameWithIndex`)
```suggestion
	{&testSplunkObservabilityMetadata[0], 0, "s0-signalfx"},
	{&testSplunkObservabilityMetadata[0], 1, "s1-signalfx"},
```